### DISCUSSION TITLE
Combobox: Allow Static and Virtual options to coexist

### DISCUSSION THREAD

**Discussion by ericwilhite** [tailwindlabs/headlessui#2753] [Ideas]
I got very exited about the new `virtual` option introduced in #2740 !

But then I saw the note that the `static` option cannot be used when using virtualization. 

We are using the Combobox (react) to power a search component very similar in style and function to [this example from the CmdK library](https://cmdk.paco.me/).

The component has thousands of options and we have built a custom (and somewhat flaky) virtualized Combobox using HeadlessUI/React and @tanstack/react-virtual. I would absolutely love to do away with that but the `static` option is a must have in order to do so.

The justification used in the PR is the entire goal is to not render items to the DOM. But since virtualization is in place that should be handled by the virtualization itself in the case of rendering the static (open) Combobox Options. In our case only ~10 options would be rendered on the component mount.

**Comment by ferrucc-io** [comment]
100% having the same issue with using the component in a cmd-k menu!

**Comment by RobinMalfait** [comment]
Hey!

Can you share a minimal reproduction repo or some code, I would love to see why the `static` prop is needed exactly for your use case.
